File: client/src/app/entities/studentpayment.ts
import {User} from './user';
import {Student} from './student';
import {Enrollment} from './enrollment';
import {DataPage} from '../shared/data-page';

export class Studentpayment {
  id: number;
  code: string;
  student: Student;
  enrollment: Enrollment;
  prevbalance: number;
  amount: number;
  balance: number;
  insno: number;
  date: string;
  description: string;
  creator: User;
  tocreation: string;
}

export class StudentpaymentDataPage extends DataPage{
    content: Studentpayment[];
}

File: server/src/main/java/bit/project/server/entity/Studentpayment.java
package bit.project.server.entity;

import lombok.Data;
import javax.persistence.*;
import java.time.LocalDate;
import javax.persistence.Id;
import java.math.BigDecimal;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Studentpayment{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private BigDecimal prevbalance;

    private BigDecimal amount;

    private BigDecimal balance;

    private Integer insno;

    private LocalDate date;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Student student;

    @ManyToOne
    private Enrollment enrollment;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    public Studentpayment(Integer id){
        this.id = id;
    }

    public Studentpayment(Integer id, String code, Student student, Enrollment enrollment){
        this.id = id;
        this.code = code;
        this.student = student;
        this.enrollment = enrollment;
    }

}